docs/001_develop/03_client-capabilities/023_custom-components/03_state-management

```diff
@@ -0,0 +1,342 @@
+
+    handleClick() {
+        this.clickCount++; // Update internal state
+        console.log(`Button clicked ${this.clickCount} times`);
+    }
+}
+```
+
+- They can be declared using @attr which provides reflection between the JavaScript property and the HTML attribute. Changes to the attribute update the property, and vice versa.
+
+```shell
+export class MyButton extends FASTElement {
+    @attr label = "Click Me"; // Declared as a linked property
+}
+
+```
+
+- The : symbol is used for property binding, allowing you to bind values directly to DOM properties instead of HTML attributes. This is useful when you want to bind complex data types or frequently changing properties that shouldn’t be reflected as attributes in the DOM.
+
+See `clickCount`
+
+```shell
+
+@customElement({
+  name: 'my-button',
+  template: html`
+    <<button> :clickCount="${x => x.clickCount}"></button>
+`;,
+  styles,
+})
+
+export class MyButton extends GenesisElemen {
+    @attr label = "Click Me"; // Linked attribute for label
+    clickCount = 0; // Internal property, not linked to an attribute
+
+    handleClick() {
+        this.clickCount++; // Update internal state
+        console.log(`Button clicked ${this.clickCount} times`);
+    }
+}
+
+```
+
+<!-- JavaScript values with type safety
+Reflect to attributes automatically
+Used for external API -->
+
+## Observables
+
+Observables are a powerful tool for creating reactive components. They allow properties to trigger reactivity within a component whenever the data changes, which is essential for dynamic and interactive elements. Observables are declared with the `@observable`.
+
+- Observable properties automatically trigger updates to any part of the template or component that depends on them. Whenever an observable property changes, the UI or other bound elements update without additional code.
+- Use `@observable` to mark a property as observable. This ensures that changes to the property automatically trigger any relevant updates in the component.
+
+```shell
+
+@customElement({
+  name: 'my-button',
+  template: html`
+    <button :clickCount="${x => x.clickCount}"></button>
+`;,
+  styles,
+})
+
+export class MyButton extends GenesisElemen {
+   @observable value = "Initial Value";
+}
+
+```
+
+- For each observable property, we have a built-in mechanism to execute code whenever the property’s value changes. This callback method is named after the observable property, with the suffix Changed.
+
+```shell
+export class MyButton extends GenesisElemen {
+  @observable value = "Initial Value";
+
+valueChanged(oldValue, newValue) {
+    console.log(`Value changed from ${oldValue} to ${newValue}`);
+  }
+}
+
+```
+
+Let’s look at an example where we create a <my-button> component with an observable count property that updates dynamically every time a button is clicked. We’ll use the `@observable` decorator and a `countChanged` callback to log each change.
+
+```shell
```

Comment:
Need to update the example according to previous comments, however this is a really clear and informative example! Nice one 🎉